src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -99,6 +101,7 @@ public boolean reduceScopeForLimitedSpace(long expectedSize)
      * which are properly serialized.
      * Caller is in charge of marking/unmarking the sstables as compacting.
      */
+    @SuppressWarnings("resource") // It is dangerous to close refs for a failed transaction
```

Comment:
we should add a larger comment detailing the specifics of what we're trying to avoid/fix here